src/core/Akka.TestKit/TestKitBase.cs

```diff
@@ -357,7 +357,33 @@ public void IgnoreNoMessages()
         /// <returns>The actor to watch, i.e. the parameter <paramref name="actorToWatch"/></returns>
         public IActorRef Watch(IActorRef actorToWatch)
         {
-            _testState.TestActor.Tell(new TestActor.Watch(actorToWatch));
+            /*
+             * Look, I know what you're thinking: wow, what kind of idiot would add this line of code?
+             * Did you know that this method is secretly asynchronous? And has been responsible for possibly dozens of
+             * racy unit tests? Why yes, yes dear reader - in fact it has!
+             *
+             * We have to ensure that the Watch completes before this method exit, so unfortunately we have to block.
+             *
+             * "Nuke the site from orbit. It's the only way to be sure."
+             */
+            WatchAsync(actorToWatch).Wait(RemainingOrDefault);
+            return actorToWatch;
+        }
+        
+        /// <summary>
+        /// Have the <see cref="TestActor"/> watch an actor and receive 
+        /// <see cref="Terminated"/> messages when the actor terminates.
+        /// </summary>
+        /// <param name="actorToWatch">The actor to watch.</param>
+        /// <returns>The actor to watch, i.e. the parameter <paramref name="actorToWatch"/></returns>
+        /// <remarks>
+        /// This method exists in order to make the asynchronous nature of the Watch method explicit.
+        /// </remarks>
+        public async Task<IActorRef> WatchAsync(IActorRef actorToWatch)
+        {
+            var taskCompletionSource = new TaskCompletionSource<bool>();
+            _testState.TestActor.Tell(new TestActor.Watch(actorToWatch, taskCompletionSource));
```

Comment:
Would it be less API breaking if we do an `Ask<T>()` with timeout here instead of passing a `TaskCompletionSource` to the test actor?

Reply:
We also need to make sure that `TestActor.Watch` and `TestActor.Unwatch` is future compatible, just in case someone uses it to message a remote actor.

Reply:
> We also need to make sure that TestActor.Watch and TestActor.Unwatch is future compatible, just in case someone uses it to message a remote actor.

Ah, for a remote `TestActor` ? Yeah good point - I don't think we have any use cases where that happens in the test suite currently but rewriting this method to just send an ACK back eliminates that as a possible issue altogether.

Reply:
Resolved